### ISSUE TITLE
Process the ECE Response file from Institutions

### ISSUE THREAD

**Issue by dheepak-aot** [bcgov/SIMS#2563]
**Describe the task**
Process the ECE response files from Institutions and provide detailed information on errors if any with the files and file records.

**Acceptance Criteria**
- [x] Process the ECE files in TEST env/Test env data and provide the details of the errors on file or file records.

**Comment by dheepak-aot** [comment]
Testing CONR_008.TXT for Location AUAA with file [AUAA_CONR_008.txt](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeEUyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3d0ced604eed042cea447aeea0f51efbd6fb4393/AUAA_CONR_008.txt)

Tested without any change in the file
Output:  All the 

<img width="761" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeE0yQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f69830a7bd8430dbcdcf9c61cb8b778d688c5b5b/image.png" alt="image.png" />

**Comment by dheepak-aot** [comment]
After updating the disbursement schedule id instead of the disbursement value id, all the ECE records belonging to SIMS were processed. 

Manually updated file[AUAA_MODIFIED_CONR_008.TXT](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMWsyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5dfbc3b85efc02ef0c14daec42dc8729f41828d7/AUAA_MODIFIED_CONR_008.TXT)


<img width="1363" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMWcyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--cce8806d00ab25dcd66badb49097ab2fff3f56d8/image.png" alt="image.png" />

<img width="853" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMW8yQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--32ad4b31cccb43b1c61a56d21cb113d82b138425/image.png" alt="image.png" />

<img width="926" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMXMyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--971fb800042b3bd77771d977e648ec6bc4658f46/image.png" alt="image.png" />

**Comment by dheepak-aot** [comment]
Testing CONR_008.TXT for Location APAK with manually modified file. 
After updating the disbursement schedule id instead of the disbursement value id, all the ECE records belonging to SIMS were processed.

<img width="1607" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM3cyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a1309ae0c3bf102a44724aa1e72403f6fc085b7b/image.png" alt="image.png" />

Summary of processing:

<img width="891" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMzQyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--fe47e3aed01cfec1aa017ad9866e9fe262eff345/image.png" alt="image.png" />

Skipped ones:
<img width="1226" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNEEyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--06a2893aef095d3ca68e32a9d75456599ea2edae/image.png" alt="image.png" />

**Comment by dheepak-aot** [comment]
Testing CONR_008.TXT for Location AUAE with manually modified file. 
Manually updated file [APAK_MODIFIED_CONR_008.TXT](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNUUyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3590219d1ab0e3ccc72a8adaad03ea9a6b0a7b7f/APAK_MODIFIED_CONR_008.TXT)


After updating the disbursement schedule id instead of the disbursement value id, the file failed to process due to file parsing error.

**Comment by dheepak-aot** [comment]
Testing CONR_008.TXT for Location AUAE with manually modified file. 
Manually updated file[AUAE_MODIFIED_1_CONR_008.TXT](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNXcyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f74b0047b6e2b4eff4a86e23159e2d41a68ced1a/AUAE_MODIFIED_1_CONR_008.TXT)


After updating the disbursement schedule id instead of the disbursement value id and removing 2 records which did not have application number, all the ECE records belonging to SIMS were processed.

Processing Summary:
<img width="859" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNTAyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0e68d2bcd34f7f477d96220461c3306eb1afe503/image.png" alt="image.png" />

Warnings: 
<img width="1198" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNTgyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1c2cf865080065fb1cea459e7629db3aa5122c3f/image.png" alt="image.png" />

Errors:
<img width="1269" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNkEyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ba7245dff8a3a7954bfe0a9ae0ccf8eddef17777/image.png" alt="image.png" />

**Comment by dheepak-aot** [comment]
Testing CONR_008.TXT for Location AUAZ with manually modified file. 
Manually updated file[AUAZ_MODIFIED_CONR_008.TXT](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN0UyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ba773e8d274149525a454a662b1dc7475c817cfc/AUAZ_MODIFIED_CONR_008.TXT)


After updating the disbursement schedule id instead of the disbursement value id, all the ECE records belonging to SIMS were processed.

<img width="1728" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN0kyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b3d4cc286c2ae0886c6703e03a519abb5a581a4c/image.png" alt="image.png" />

Process Summary: 
<img width="928" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN1EyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f54f787b5d82f28e053e5a347f287fe19c5ada85/image.png" alt="image.png" />

**Comment by dheepak-aot** [comment]
Query to get the disbursement schedule id of a disbursement value id and joining the application.

```sql
select dv.id as disbursement_value_id , dv.disbursement_schedule_id, a.application_number  from sims.disbursement_values dv inner join 

sims.disbursement_schedules ds on dv.disbursement_schedule_id = ds.id inner join sims.student_assessments sa on ds.student_assessment_id = sa.id 

inner join sims.applications a on a.current_assessment_id = sa.id where a.application_number  = $application_number;
```

**Comment by dheepak-aot** [comment]
Here is the summary of the test results

<img width="1233" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN3M2QWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--106143109300ab7a26ae793f680d258c73f50a8e/image.png" alt="image.png" />

**Comment by HRAGANBC** [comment]
@dheepak-aot @cditcher Thanks so much Dheepak, this was hugely helpful. 

One question for you: when making the fix to manually align with Disbursement Schedule ID and Award IDs, it looks like you adjust the file itself to include the disbursement schedule ID, is that correct? 

If so, that is fine for the above testing purposes, but we need to keep the Award ID field in the ECE file exactly as it is currently, and change the code which processes that file so that SIMS expects the Award ID when processing the files. This is reflected in ticket #2538. I just want to make sure that's OK with you, and you made changes to the file purely for testing purposes.

**Comment by dheepak-aot** [comment]
Yes Hugh. I adjust the file content replacing `disbursement shedule id` with `disbursement value id`.
I see the required code changes to fix this is captured in #2538 and we can bring it to grooming based on priority and eventually implement the code changes.
